at_onboarding_flutter/example/lib/main.dart

```diff
@@ -102,36 +105,78 @@ class _MyAppState extends State<MyApp> {
             body: Builder(
               builder: (context) => Center(
                 child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
                   children: [
                     ElevatedButton(
                       onPressed: () async {
                         var preference = await futurePreference;
                         setState(() {
                           atClientPreference = preference;
                         });
-                        Onboarding(
+                        final result = await AtOnboarding.onboard(
                           context: context,
-                          atClientPreference: atClientPreference!,
-                          domain: AtEnv.rootDomain,
-                          rootEnvironment: AtEnv.rootEnvironment,
-                          appAPIKey: AtEnv.appApiKey,
-                          appColor: Theme.of(context).primaryColor,
-                          onboard: (value, atsign) {
-                            _logger.finer('Successfully onboarded $atsign');
-                          },
-                          onError: (error) {
-                            _logger.severe('Onboarding throws $error error');
-                          },
-                          nextScreen: const HomeScreen(),
+                          config: AtOnboardingConfig(
+                            atClientPreference: atClientPreference!,
+                            domain: AtEnv.rootDomain,
+                            rootEnvironment: AtEnv.rootEnvironment,
+                            appAPIKey: AtEnv.appApiKey,
+                          ),
                         );
+                        switch (result.status) {
+                          case AtOnboardingResultStatus.success:
+                            final atsign = result.atsign;
+                            Navigator.push(
+                                context,
+                                MaterialPageRoute(
+                                    builder: (_) => const HomeScreen()));
+                            break;
+                          case AtOnboardingResultStatus.error:
+                            // TODO: handle onboard failure

```

Comment:
@sonle-geekyants Can we add an error snackbar here, to show something went wrong while onboarding.